src/main/java/com/glencoesoftware/omero/ms/image/region/ZarrPixelBuffer.java

```diff
@@ -669,11 +656,16 @@ public Dimension getTileSize() {
     @Override
     public List<List<Integer>> getResolutionDescriptions() {
         try {
-            int[][] shapes = getShapes();
+            int resolutionLevels = getResolutionLevels();
             List<List<Integer>> resolutionDescriptions =
                     new ArrayList<List<Integer>>();
-            for (int[] shape : shapes) {
-                resolutionDescriptions.add(Arrays.asList(shape[4], shape[3]));
+            int sizeX = pixels.getSizeX();
+            int sizeY = pixels.getSizeY();
+            for (int i = 0; i < resolutionLevels; i++) {
+                double scale = Math.pow(2, i);
```

Comment:
Are the resolution levels always successive powers of 2 from each other?

Reply:
Yes, at least as far as we use them. It's not explicitly defined but should be in the future:

* https://github.com/zarr-developers/zarr-specs/issues/50